File: 3762.minimum-operations-to-equalize-subarrays.cpp
#
# @lc app=leetcode id=3762 lang=cpp
#
# [3762] Minimum Operations to Equalize Subarrays
#
# @lc code=start
class Solution {
public:
    vector<long long> minOperations(vector<int>& nums, int k, vector<vector<int>>& queries) {
        vector<long long> result;
        
        for (auto& query : queries) {
            int l = query[0];
            int r = query[1];
            
            // If subarray has only one element, no operations needed
            if (l == r) {
                result.push_back(0);
                continue;
            }
            
            // Check if all elements have the same remainder modulo k
            long long remainder = nums[l] % k;
            bool possible = true;
            
            vector<long long> coefficients;
            for (int i = l; i <= r; i++) {
                if (nums[i] % k != remainder) {
                    possible = false;
                    break;
                }
                // Each nums[i] = remainder + coefficient * k
                long long coefficient = (nums[i] - remainder) / k;
                coefficients.push_back(coefficient);
            }
            
            if (!possible) {
                result.push_back(-1);
                continue;
            }
            
            // Sort to find median
            sort(coefficients.begin(), coefficients.end());
            
            // Use median as target coefficient
            long long median = coefficients[coefficients.size() / 2];
            
            // Calculate total operations
            long long ops = 0;
            for (long long coef : coefficients) {
                ops += abs(coef - median);
            }
            
            result.push_back(ops);
        }
        
        return result;
    }
};
# @lc code=end

File: 3762.minimum-operations-to-equalize-subarrays.java
#
# @lc app=leetcode id=3762 lang=java
#
# [3762] Minimum Operations to Equalize Subarrays
#

# @lc code=start
class Solution {
    public long[] minOperations(int[] nums, int k, int[][] queries) {
        int n = nums.length;
        int q = queries.length;
        long[] ans = new long[q];
        for (int idx = 0; idx < q; ++idx) {
            int l = queries[idx][0], r = queries[idx][1];
            int mod = nums[l] % k;
            boolean ok = true;
            for (int i = l + 1; i <= r; ++i) {
                if (nums[i] % k != mod) {
                    ok = false;
                    break;
                }
            }
            // Edge case: single-element subarray
            if (!ok) {
                ans[idx] = -1;
                continue;
            }
            int len = r - l + 1;
            if (len == 1) {
                ans[idx] = 0;
                continue;
            }
            int[] normalized = new int[len];
            for (int i = 0; i < len; ++i) {
                normalized[i] = (nums[l + i] - mod) / k;
            }
            java.util.Arrays.sort(normalized);
            int median = normalized[len / 2];
            long sum = 0;
            for (int v : normalized) {
                sum += Math.abs(v - median);
            }
            ans[idx] = sum;
        }
        return ans;
    }
}
# @lc code=end